### PR TITLE
ghosting while in crit succumbs you first so you can re-enter corpse

### LINKED ISSUE #7065: Ghosting while dead/dying
Click ghost while in crit, do not confirm ghosting. Wait until you die then click yes. You are now unable to reenter body as if you went catatonic

On an unrelated note, ghosting while in crit should kill you as if last whispering and allow you to re enter body.

### PR DESCRIPTION
**Pull request by Intigracy**
fixes #7065

tested-stamp

### CODE REVIEW

**Comment by Probe1** [comment]
thats intended

**Comment by Intigracy** [comment]
No, it's intended for if you ghost while alive as punishment for ghosting while handcuffed or something.

If you're already in crit then there's nothing wrong with being allowed to enter your body again after ghosting out of it before succumbing. It's extremely unintuitive as-is.

**Comment by 9600bauds** [comment]
I think you could word this better to note that this makes you succumb before ghosting you

**Comment by Intigracy** [comment]
there

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -296,7 +296,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	set name = "Ghost"
 	set desc = "Relinquish your life and enter the land of the dead."
 
-	if(stat == DEAD)
+	if(src.health < 0 && src.health > -95.0) //crit people
+		succumb()
+		sleep(5)
```
Comment (ghost) [inline code/modules/mob/dead/observer/observer.dm:301]: Is this a good idea?

**Comment by 9600bauds** [comment]
:+1:

File: code/modules/mob/dead/observer/observer.dm (hunk 1 above)
Comment (Intigracy) [inline code/modules/mob/dead/observer/observer.dm:301]: It's to make sure it doesn't continue with the else chunk first, which is what happened to me during testing.

It's half a second and barely noticable.

Comment (ghost) [inline code/modules/mob/dead/observer/observer.dm:301]: what

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -296,7 +296,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+		ghostize(1)
+	else if(stat == DEAD)
 		ghostize(1)
 	else
```
Comment (Intigracy) [inline code/modules/mob/dead/observer/observer.dm:305]: no sleep = this gets ran before you finish succumbing

File: code/modules/mob/dead/observer/observer.dm [hunk 3]
```diff
@@ -296,7 +296,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
 		var/response = alert(src, "Are you -sure- you want to ghost?\n(You are alive. If you ghost, you won't be able to play this round for another 30 minutes! You can't change your mind so choose wisely!)","Are you sure you want to ghost?","Ghost","Stay in body")
```
Comment (Intigracy) [inline code/modules/mob/dead/observer/observer.dm:306]: this gets ran before you finish succumbing = you're still going to end up ghosted and unable to enter corpse

**Comment by Intigracy** [comment]
no sleep = you're still going to end up ghosted and unable to enter corpse

**Comment by ghost** [comment]
They're called else statements for a reason.

File: code/modules/mob/dead/observer/observer.dm (hunk 2 above)
Comment (ghost) [inline code/modules/mob/dead/observer/observer.dm:305]: > else

File: code/modules/mob/dead/observer/observer.dm (hunk 3 above)
Comment (ghost) [inline code/modules/mob/dead/observer/observer.dm:306]: > else

**Comment by Intigracy** [comment]
Test it without the sleep if you don't believe me then.

**Comment by gbasood** [comment]
Or you could use spawn() so it's non-blocking
Regardless of what Celt may be talking about, that's important

**Comment by Intigracy** [comment]
The blocking of the proc is intentional, using spawn() would still have the same issue.

**Comment by Probe1** [comment]
emojis were a mistake

**Comment by Intigracy** [comment]
It lets you meme without even having to post anymore.

**Comment by Intigracy** [comment]
You can merge this now bb.

File: code/modules/mob/dead/observer/observer.dm (hunk 2 above)
Comment (Exxion) [inline code/modules/mob/dead/observer/observer.dm:305]: Intigracy what the fuck are you talking about there is no way for both that `sleep()` and this code block to execute in the same call
I think I know what you actually mean, but you're explaining it incorrectly

**Comment by FudgePucker** [comment]
All the reactions are missing are :ok_hand: :joy: and :100: 
MMM THATS SOME GOOD SHIT :joy: :joy: :joy: :ok_hand: :ok_hand: :100:

**Comment by Probe1** [comment]
its not a bug bruh.

**Comment by ihadtoregisterforthis** [comment]
Reactions should be customizable by the owner

**Comment by Intigracy** [comment]
It's not a bug, it's annoying.

**Comment by Probe1** [comment]
I'll agree with that.

**Comment by clusterfack** [comment]
Quality of death

**Comment by Probe1** [comment]
punch it coderman of collaboration.

File: code/modules/mob/dead/observer/observer.dm (hunk 3 above)
Comment (gbasood) [inline code/modules/mob/dead/observer/observer.dm:306]: Due to how else blocks work, yes, this should never be reached if it reaches the code you added in first.
Don't make a workaround, figure out what's happening to cause this instead.

Comment (ghost) [inline code/modules/mob/dead/observer/observer.dm:306]: @Clusterfack why the fuck did you merge this?

Comment (Intigracy) [inline code/modules/mob/dead/observer/observer.dm]: @DrCelt why the fuck haven't you fixed ventcrawling for mobs?

File: code/modules/mob/dead/observer/observer.dm (hunk 1 above)
Comment (ComicIronic) [inline code/modules/mob/dead/observer/observer.dm:301]: What the shit are you even talking about here

**Pull request merged**